makefiles/tools/serial.inc.mk

```diff
@@ -1,9 +1,41 @@
 # set default port depending on operating system
+# If the board does not provide a configuration, we assume that the serial port
+# is some kind of USB to serial adapter and try to search for a device with a
+# serial number matching the variable USB_UART_ID, falling back to
+# DEBUG_ADAPTER_ID if USB_UART_ID is not given.
+# If no serial number is given, then we pick the first USB tty we can find
 OS := $(shell uname)
-ifeq ($(OS),Linux)
-  PORT ?= $(PORT_LINUX)
-else ifeq ($(OS),Darwin)
-  PORT ?= $(PORT_DARWIN)
+ifeq (,$(PORT))
+  ifeq (,$(USB_UART_ID))
+    USB_UART_ID := $(DEBUG_ADAPTER_ID)
+  endif
```

Comment:
Any reason, why you decidet against

```makefile
USB_UART_ID ?= $(DEBUG_ADAPTER_ID)
```

here and in some other instances below?

Reply:
I wanted to make sure that `USB_UART_ID= make term` behave the same as `unset USB_UART_ID; make term`

Reply:
or if it was assigned an empty string from one of the board makefiles or other build system file, that it would still get a fallback value